src/EFCore.MySql/Query/Expressions/Internal/MySqlMatchExpression.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.EntityFrameworkCore.Utilities;
+using Pomelo.EntityFrameworkCore.MySql.Query.ExpressionVisitors.Internal;
+
+namespace Pomelo.EntityFrameworkCore.MySql.Query.Expressions.Internal
+{
+    public enum MySqlMatchExpressionSearchMode
+    {
+        WithQueryExpansion,
+        InNaturalLanguageMode,
+        InNaturalLanguageModeWithQueryExpansion,
+        InBooleanMode
+    }
```

Comment:
Use this enum for the one public match method and rename it accordingly (e.g. to `MySqlMatchSearchMode`).